src/main/java/edu/wpi/grip/core/events/StepAddedEvent.java

```diff
@@ -0,0 +1,51 @@
+package edu.wpi.grip.core.events;
+
+import com.google.common.base.MoreObjects;
+import edu.wpi.grip.core.Socket;
+import edu.wpi.grip.core.Step;
+
+/**
+ * An event that occurs when a new step is added to the pipeline.  This is triggered by the user adding a step with the
+ * GUI.
+ */
+public class StepAddedEvent {
+    private Step step;
+    private int index;
+
+    /**
+     * @param step  The step being added
+     * @param index The index that the step should be added at
+     */
+    public StepAddedEvent(Step step, int index) {
+        this.step = step;
+        this.index = index;
+    }
+    /**
+     * @param step The step being added to the end of the pipeline
+     */
+    public StepAddedEvent(Step step) {
+        this(step, -1);
+    }
+
+    /**
+     * @return The step being added.
+     */
+    public Step getStep() {
+        return this.step;
+    }
+
+    /**
+     * @return The index that the step should be added at, or -1 to append it to the end of the pipeline.
+     */
+    public int getIndex() {
```

Comment:
If the only usage of getIndex is to check whether or not the step should be added to the end then it would be better to call it something like `shouldBeLastStep`
Or, have the step add itself to the list.
It would help to be as verbose as possible with the logic.